tests/test-media-item-mutations.php

```diff
@@ -0,0 +1,803 @@
+	 * get an error back from the source because they are required.
+	 *
+	 * @source WPGraphql - createMediaItemInput!
+	 * @access public
+	 * @return void
+	 */
+	public function testCmiNoInput() {
+
+		/**
+		 * Set up the createMediaItem mutation
+		 */
+		$mutation = '
+		mutation createMediaItem( $input: createMediaItemInput! ){
+		  createMediaItem(input: $input){
+		    clientMutationId
+		    mediaItem{
+		      id
+		    }
+		  }
+		}
+		';
+
+		$empty_variables = '';
+		$actual = do_graphql_request( $mutation, 'createMediaItem', $empty_variables );
+		$this->assertArrayHasKey( 'errors', $actual );
+	}
+
+	/**
+	 * Set the current user to subscriber (someone who can't create posts)
+	 * and test whether they can create posts
+	 *
+	 * @source wp-content/plugins/wp-graphql/src/Type/MediaItem/MediaItemCreate.php:61
+	 * @access public
+	 * @return void
+	 */
+	public function testCmiAsSubscriber() {
+		wp_set_current_user( $this->subscriber );
+		$actual = $this->createMediaItemMutation();
+		$this->assertArrayHasKey( 'errors', $actual );
+	}
+
+	/**
+	 * Set the current user to subscriber (someone who can't create posts)
+	 * and test whether they can create posts with someone else's id
+	 *
+	 * @source wp-content/plugins/wp-graphql/src/Type/MediaItem/MediaItemCreate.php:61
+	 * @access public
+	 * @return void
+	 */
+	public function testCmiOtherAuthor() {
+
+		/**
+		 * Set up the createMediaItem mutation
+		 */
+		$mutation = '
+		mutation createMediaItem( $input: createMediaItemInput! ){
+		  createMediaItem(input: $input){
+		    clientMutationId
+		    mediaItem{
+		      id
+		    }
+		  }
+		}
+		';
+
+		/**
+		 * Set the createMediaItem mutation input variables
+		 */
+		$variables = [
+			'input' => [
+				'filePath'         => $this->filePath,
+				'fileType'         => $this->fileType,
+				'clientMutationId' => $this->clientMutationId,
+				'title'            => $this->title,
+				'description'      => $this->description,
+				'altText'          => $this->altText,
+				'parentId'         => $this->parentId,
+				'caption'          => $this->caption,
+				'commentStatus'    => $this->commentStatus,
+				'date'             => $this->date,
+				'dateGmt'          => $this->dateGmt,
+				'slug'             => $this->slug,
+				'status'           => $this->status,
+				'pingStatus'       => $this->pingStatus,
+				'authorId'         => $this->admin,
+			],
+		];
+
+		wp_set_current_user( $this->author );
+		$actual = do_graphql_request( $mutation, 'createMediaItem', $variables );
+		$this->assertArrayHasKey( 'errors', $actual );
+	}
+
+	/**
+	 * Test whether we need to include the file.php file
+	 * from the wp-admin
+	 *
+	 * @source wp-content/plugins/wp-graphql/src/Type/MediaItem/Mutation/MediaItemCreate.php:76
+	 */
+	public function testCmiRequireFilePhp() {
+
+		/**
+		 * download_url_file will equal true if the file is already included
+		 */
+		$download_url_file = require_once( ABSPATH . 'wp-admin/includes/file.php' );
```

Comment:
so, I would actually test running a mutation where this file was already included (like if another plugin included it already) to make sure that the duplicate includes don't cause an issue. 

So, something like: 
```
require_once( ABSPATH . 'wp-admin/includes/file.php' );
$mutation = 'define your mutation here';
$variables = [ 'Define your variables here' ];
$actual = do_graphql_request( $mutation, 'createMediaItemWithDuplicateFileInclude', $variables );
// Then make your assertion(s) that the mutation processed without error. . .
```
If there was an error due to the functions in that file being defined multiple times or whatever, we'll find out here.

But right now the tests are _only_ testing whether PHP's `require_once` actually does require the file properly, which is fine, but we want to also make sure that requiring that file when it's already been required by another part of the code doesn't cause issues.